src/query/block/column.go

```diff
@@ -237,19 +261,25 @@ func (m *columnBlockSeriesIter) SeriesCount() int {
 	return len(cols[0].Values)
 }
 
-func (m *columnBlockSeriesIter) Next() bool {
-	m.idx++
-	return m.idx < m.SeriesCount()
+func (m *columnBlockSeriesIter) Err() error {
+	return nil
 }
 
-func (m *columnBlockSeriesIter) Current() (Series, error) {
-	cols := m.columns
-	values := make([]float64, len(cols))
-	for i := 0; i < len(cols); i++ {
-		values[i] = cols[i].Values[m.idx]
+func (m *columnBlockSeriesIter) Next() bool {
+	m.idx++
+	next := m.idx < m.SeriesCount()
+	if next {
+		cols := m.columns
+		for i := 0; i < len(cols); i++ {
+			m.values[i] = cols[i].Values[m.idx]
+		}
 	}
 
-	return NewSeries(values, m.seriesMeta[m.idx]), nil
+	return next
+}
+
+func (m *columnBlockSeriesIter) Current() Series {
+	return NewSeries(m.values, m.seriesMeta[m.idx])
```

Comment:
Just checking--does our usage of the iterator ensure that we never use the `values` slice after another call to `Next`?

i.e. nothing like:

```
firstValues := iter.Current()
iter.Next()
nextValues := iter.Current()
``` 

Maybe worth adding a comment to the `Next` method saying that it invalidates previously returned arrays, and that users should copy if they need to keep that data around?

Reply:
Added a comment to reflect this. I think failing anything else coming up with more priority, my next task would include adding pooling to these iterators, which should make it safer.

Also this one will only be used by LegacyBlockProcessing, which shouldn't be around much longer and is not the default, so less worried about it.

Reply:
Actually, just going to copy the array instead as this would be a nightmare to debug if it happens :p